lib/install.js

```diff
@@ -0,0 +1,82 @@
+const { resolve } = require('path');
+const fs = require('fs-extra');
+const assert = require('bsert');
+const validate = require('validate-npm-package-name');
+const { format } = require('prettier');
+
+const { checkExistence } = require('./helpers');
+
+function filterArgs() {
+  // remove node, path, and command args
+  const args = process.argv.slice(3);
+  let names = args.filter(name => name[0] !== '-');
+  return names;
+}
+
+module.exports = (config, logger) => async (name, cmd) => {
+  let bpanelConfig, configPath;
+
+  try {
+    const localPluginsDir = resolve(config.prefix, 'local_plugins');
+    configPath = resolve(config.prefix, 'config.js');
+
+    assert(fs.existsSync(configPath), 'Could not find bPanel config file');
+    if (cmd.local)
+      assert(
+        fs.existsSync(localPluginsDir),
+        'Local plugins directory does not exist. ' +
+          'Ensure bPanel has been initiated and --prefix option has been set correctly.'
+      );
+
+    bpanelConfig = require(configPath);
+
+    const names = filterArgs();
+
+    const configList = cmd.local ? bpanelConfig.localPlugins : bpanelConfig.plugins;
+    const startLength = configList.length;
+
+    assert(names.indexOf(name) > -1, 'There was a problem parsing args');
+
+    for (let i = 0; i < names.length; i++) {
+      let includeConfig = true;
+      // validate the name
+      const validator = validate(names[i]);
+      const errorStr = `Skipping ${names[i]}:`;
+      if (validator.warnings) {
+        includeConfig = false;
+        logger.error(`${errorStr} ${validator.warnings.join(', ')}`);
+        continue;
+      }
+
+      if (validator.errors) {
+        includeConfig = false
+        logger.error(`${errorStr} ${validator.errors.join(', ')}`);
+        continue;
+      }
+
+      const exists = checkExistence(names[i], configList, cmd.local);
+
+      if (exists) {
```

Comment:
A future person refactoring may introduce an error (wouldn't be fatal in this case) if `exists` returns as `false` and then `exists.message` is referenced. It seems like a leaky abstraction, as you need to know the logic happening inside of `checkExistence` to be able to refactor this code